pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -266,6 +266,7 @@ protected void createColumnV1Indices(String column)
     int dictionaryElementSize = 0;
 
     Object sortedArray;
+    final int cardinality = 1;
```

Comment:
Prior to this change, the callers to construct ColumnIndexCreationInfo() had to provide an array of unique values as one of the constructor arguments. Internally, ColumnIndexCreationInfo computed the cardinality as array.length.

With this change, I have introduced an additional argument to the ctor of ColumnIndexCreationInfo -- cardinality.

If you see the lines below, the array of unique values is computed as a one-element array consisting of the default value of the column being added. Since the size of this array is 1, I am hard-setting the cardinality to 1 and passing that value in the constructor. I did not want to hard-code the value in the call to constructor.